firewood/src/db.rs

```diff
@@ -411,6 +412,10 @@ impl api::Db for Db {
     ) -> Result<Self::Proposal, api::Error> {
         block_in_place(|| self.new_proposal(batch)).map_err(Into::into)
     }
+
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
```

Comment:
🤔 I think I would prefer to see this inline even if it's repeated. But I'm definitely curious to see where this is needed.